lib/rubocop/cop/rspec/message_spies.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that message expectations are set using spies.
+      #
+      # This cop can be configured in your configuration using the
+      # `EnforcedStyle` option and supports `--auto-gen-config`.
+      #
+      # @example `EnforcedStyle: have_received`
+      #
+      #   # bad
+      #   expect(foo).to receive(:bar)
+      #
+      #   # good
+      #   expect(foo).to have_received(:bar)
+      #
+      # @example `EnforcedStyle: receive`
+      #
+      #   # bad
+      #   expect(foo).to have_received(:bar)
+      #
+      #   # good
+      #   expect(foo).to receive(:bar)
+      #
+      class MessageSpies < Cop
+        include RuboCop::RSpec::SpecOnly, ConfigurableEnforcedStyle
+
+        MSG_RECEIVE = 'Prefer `receive` for setting message ' \
+          'expectations.'.freeze
+        MSG_HAVE_RECEIVED = 'Prefer `have_received` for setting message ' \
+          'expectations. You will need to set up `%s` as a spy, using either ' \
```

Comment:
Small nit pick: How about "Setup `%s` as a spy using `allow` or `instance_spy`". This seems better for two reasons:

1. A bit more concise (but I don't feel that strongly about this so if you disagree then don't feel a need to change)
2. Defaults to encouraging `instance_spy` instead of `spy` (same benefit as using `instance_double` instead of `double`)